### PR TITLE
Map: Don't show Tooltip for countries outside the projection

### PR DESCRIPTION
**Pull request by MarcelGerber**
Until now, Grapher showed Tooltips for countries outside the current projection:
image

This is very inconsistent, for multiple reasons:
* The country for which the tooltip is shown is not even highlighted with a thicker border (the cursor changes to `pointer`, though).
* Tooltips were only shown using the "close to center" logic (`onMouseMove`), not the "inside country" logic (`onMouseEnter`). Therefore, a tooltip is not shown for large countries even if your cursor is inside it (Brazil with "Africa" projection is a good example).

This PR changes it so a tooltip is never shown for countries outside the current projection.

cc @danielgavrilov

### CODE REVIEW

File: charts/MapLegend.tsx [hunk 1]
```diff
@@ -708,7 +708,6 @@ export class MapLegendView extends React.Component<MapLegendViewProps> {
                         onMouseLeave={onMouseLeave}
                     />
                 )}
-                />}
```
Comment (MarcelGerber) [inline charts/MapLegend.tsx]: This is not directly related to this PR, but this is a bogus closing tag/brace.
I'm surprised no linter caught this.

**Comment by danielgavrilov** [comment]
Awesome, thanks for catching these consistency issues @MarcelGerber!

File: charts/MapLegend.tsx (hunk 1 above)
Comment (danielgavrilov) [reply charts/MapLegend.tsx]: Ha, I'm surprised too! It seems it's valid JSX to add as many of those closing tags you want.

Comment (MarcelGerber) [reply charts/MapLegend.tsx]: Yeah, they were just added to the the DOM, so they were considered HTML at that point. Thankfully, it was in a place where the browser just ignored them.
